src/lib/metricsMgr/MetricsManager.cpp

```diff
@@ -56,84 +94,155 @@ void MetricsManager::accum(const std::string& srv, const std::string& subServ, c
-    std::map<std::string, std::map<std::string, int>*>* servMap = ix->second;
+  sem_wait(&sem);
+
+  //
+  // Three iterators needed to iterate over the 'triple-map' metrics:
+  //   serviceIter      to iterate over all services
+  //   subServiceIter   to iterate over all sub-services of a service
+  //   metricIter       to iterate over all metrics of a sub-service
+  //
+  std::map<std::string, std::map<std::string, std::map<std::string, int>*>*>::iterator  serviceIter;
+  std::map<std::string, std::map<std::string, int>*>::iterator                          subServiceIter;
+  std::map<std::string, int>::iterator                                                  metricIter;
+  JsonHelper                                                                            top;
+  JsonHelper                                                                            services;
+
+  for (serviceIter = metrics.begin(); serviceIter != metrics.end(); ++serviceIter)
+  {
+    JsonHelper                                          subServiceTop;
+    JsonHelper                                          jhSubService;
+    std::string                                         service        = serviceIter->first;
+    std::map<std::string, std::map<std::string, int>*>* servMap        = serviceIter->second;
 
-    for(std::map<std::string, std::map<std::string, int>*>::iterator jx = servMap->begin();
-        jx != servMap->end();
-        ++jx)
+    for (subServiceIter = servMap->begin(); subServiceIter != servMap->end(); ++subServiceIter)
     {
-      // For all metrics...
-      JsonHelper jhMetrics;
-
-      std::string subServ                   = jx->first;
-      std::map<std::string, int>* metricMap = jx->second;
+      JsonHelper                  jhMetrics;
+      std::string                 subServ    = subServiceIter->first;
+      std::map<std::string, int>* metricMap  = subServiceIter->second;
 
-      for(std::map<std::string, int>::iterator kx = metricMap->begin();
-          kx != metricMap->end();
-          ++kx)
+      for (metricIter = metricMap->begin(); metricIter != metricMap->end(); ++metricIter)
       {
-        // ... print it!
-        std::string metric = kx->first;
-        int value          = kx->second;
+        std::string  metric = metricIter->first;
+        int          value  = metricIter->second;
 
         jhMetrics.addNumber(metric, value);
-
       }
 
-      jhSubserv.addRaw(subServ, jhMetrics.str());
+      jhSubService.addRaw(subServ, jhMetrics.str());
     }
 
-    jhServ.addRaw(service, jhSubserv.str());
+    subServiceTop.addRaw("subservs", jhSubService.str());
+    services.addRaw(service, subServiceTop.str());
+  }
+
+  top.addRaw("services", services.str());
+  sem_post(&sem);
+  return top.str();
+}
+
+
+
+/* ****************************************************************************
+*
+* totalTimeInTransactionAdd - 
+*/
+void MetricsManager::totalTimeInTransactionAdd(struct timeval& start, struct timeval& end)
```

Comment:
Actually, it may not come back. Not if we count microseconds in one single `long long`.
Something like this:

    microSecondDelta = (end.tv_sec - start.tv_sec) * 1000000 + (end.tv_usec - start.tv_usec);

And no function needed for it.